core/modules/global/ResourceUnit.cc

```diff
@@ -159,26 +169,33 @@ ResourceUnit::_setFromPath(std::string const& path) {
             _unitType = GARBAGE;
             return;
         }
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
+        if(t.token().empty()) { _unitType = GARBAGE; return; }
         _chunk = t.tokenAsInt();
         _ingestLeafAndKeys(t.token());
     } else if(rTypeString == prefix(CQUERY)) {
         // Import as chunk query
         _unitType = CQUERY;
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
         _db = t.token();
         if(_db.empty()) {
             _unitType = GARBAGE;
             return;
         }
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
+        if(t.token().empty()) { _unitType = GARBAGE; return; }
         _chunk = t.tokenAsInt();
         _ingestLeafAndKeys(t.token());
 
     } else if(rTypeString == prefix(RESULT)) {
         _unitType = RESULT;
+        if(_markGarbageIfDone(t)) { return; }
         t.next();
         _hashName = t.token();
+        if(_hashName.empty()) { _unitType = GARBAGE; return; }
```

Comment:
Hmm, it seems this could be cleaner/simpler...  What if you had a combined next + garbage checker/setter helpers, something like:

    bool NextTokenValidated(string* outToken, bool emptyAllowed=true);
    bool NextTokenValidated(int* outToken);

    const bool ALLOW_EMPTY = true;
    const bool DISALLOW_EMPTY = false;

or some such.  Then a lot of these three or four line clauses could be collapsed to single lines like:

    if (!NextTokenValidated(&tokenDest, DISALLOW_EMPTY)) { return; }

(and yeah, I like using pointers instead of refs for output parameters, since the then required "&" at the call site makes it obvious what's going on.  You do have to check 'em for null then in the helper though.  I think it's worth it for code readability, YMMV...)